framework/meta-lib/src/tools/wasm_extractor/extractor_tests.rs

```diff
@@ -427,70 +434,77 @@ pub mod tests {
     fn test_empty_with_mem_grow() {
         if let Ok(content) = Parser::new().parse_bytes(None, EMPTY_WITH_MEM_GROW.as_bytes()) {
             let wasm_info =
-                populate_wasm_info(Path::new(""), content.to_vec(), false, &None, Vec::new())
+                populate_wasm_info(Path::new(""), &content, false, None, &HashMap::new())
                     .expect("Unable to parse WASM content.");
-            assert!(wasm_info.memory_grow_flag);
-            assert!(!wasm_info.report.has_allocator);
+            assert!(wasm_info.report.memory_grow_flag);
+            assert!(!wasm_info.report.code.has_allocator);
             assert_eq!(
                 PanicReport::WithoutMessage.to_string(),
-                wasm_info.report.has_panic.to_string()
+                wasm_info.report.code.has_panic.to_string()
             );
         }
     }
 
     #[test]
     fn test_empty_with_fail_allocator() {
         if let Ok(content) = Parser::new().parse_bytes(None, EMPTY_WITH_FAIL_ALLOCATOR.as_bytes()) {
-            let wasm_info =
-                populate_wasm_info(Path::new(""), content.to_vec(), false, &None, Vec::new())
-                    .expect("Unable to parse WASM content.");
-            assert!(!wasm_info.memory_grow_flag);
-            assert!(wasm_info.report.has_allocator);
+            let wasm_info = populate_wasm_info(
```

Comment:
populate_wasm_info has a lot of arguments, sounds like a WasmInfo object with a builder pattern might make it clearer